HappyNewYear.cpp

```diff
@@ -35,20 +34,32 @@ int main()
   tm *const pTInfo = std::localtime(&t);
   int *p=NULL;
 
-  if (pTInfo->tm_year + 1900 == 2022) 
+  if (pTInfo->tm_year + 1900 == 2022)
     {
         cout<<"Happy New year!\n";
-        
+
         if (funAtWork && enjoyInAmadeus) cout<<"It will be super cool :)\n";
-            
+
         srand(time(0));
         noMoreCovid=rand() %2; // mostly random, as the covid forecasts
-       
-        if (noMoreCovid || vaccinationDone )    cout<<"Book your next vacation flight NOW\n";
+
+        cout<< "Are you fully vaccinated now? Answer 0 for No or your favorite number for Yes : ";
+        cin>> vaccinationDone; cin.clear();
+        if ( vaccinationDone > 1 ) vaccinationDone = true;
```

Comment:
Shouldn't it be >= 1?

Reply:
true is 1, false is 0. If user enters 0 or 1 all is fine. We just have to test greater than 1 to set it to true. On top of that int is unsigned so negative values not to be tested.